src/box/memtx_tree.cc

```diff
@@ -2271,6 +2271,10 @@ struct tree_read_view_iterator {
 	 * is exhausted - pagination relies on it.
 	 */
 	struct memtx_tree_data<USE_HINT> *last;
+	/** The last element of the iteration for EQ and REQ iterators. */
+	struct memtx_tree_data<USE_HINT> *exhaust_after;
```

Comment:
`end` may be a better name (but then we'd have to make it point to the next tuple after the last one to match the STL naming conventions)